src/MapboxStyleParser.ts

```diff
@@ -277,30 +295,30 @@ export class MapboxStyleParser implements StyleParser {
    * @param layout A Mapbox layout
    * @return A GeoStylerStyle-IconSymbolizer
    */
-  getIconSymbolizerFromMapboxLayer(paint: any, layout: any): IconSymbolizer | undefined {
+  getIconSymbolizerFromMapboxSymbolLayer(paint: SymbolPaint, layout: SymbolLayout): IconSymbolizer | undefined {
     const symbolizer: IconSymbolizer = {
       kind: 'Icon',
-      allowOverlap: layout?.['icon-allow-overlap'],
-      anchor: layout?.['icon-anchor'],
+      allowOverlap: layout?.['icon-allow-overlap'] as IconSymbolizer['allowOverlap'],
+      anchor: layout?.['icon-anchor'] as IconSymbolizer['anchor'],
```

Comment:
Sorry for being late to the party, but this is important: Use caution when type-casting layout or paint properties! They can be expressions instead of literals. While it may help now to stop tsc from complaining, it will bite you as soon as the layout or paint property contains an expression.

Reply:
I guess my comment in this PR was very misleading. I meant that when we add the parsing of expressions we can use these casts as identifier where we have to parse expressions. So i 100% agree with you on this.